### PR TITLE
[WIP] Adding loader for TAU Urban Acoustic Scenes 2020 Mobile dataset

### LINKED ISSUE #41: Adding TAU Urban Acoustic Scenes 2020 Mobile
[Development](https://zenodo.org/record/3819968) and [Evaluation](https://zenodo.org/record/3685828) datasets.

### PR DESCRIPTION
**Pull request by pzinemanas**
### Description

Please include the following information at the top level docstring for the dataset's module mydataset.py:

- [x] Describe annotations included in the dataset
- [x] Indicate the total duration of the dataset in hours, and (optionally) also list the number of individual files
- [x] Mention the origin of the dataset (e.g. creator, institution)
- [x] Describe the type of audio included in the dataset
- [x] Indicate any relevant papers related to the dataset
- [x] Include a description about how the data can be accessed and the license it uses (if applicable)

#### Dataset loaders checklist:

- [x] Create a script in `scripts/`, e.g. `make_my_dataset_index.py`, which generates an index file.
- [x] Run the script on the canonical version of the dataset and save the index in `soundata/indexes/` e.g. `my_dataset_index.json`.
- [x] Create a module in soundata, e.g. `soundata/my_dataset.py`
- [x] Create tests for your loader in `tests/`, e.g. `test_my_dataset.py`
- [x] Add your module to `docs/source/soundata.rst` and `docs/source/quick_reference.rst`
- [x] Run `tests/test_full_dataset.py` on your dataset.

#### TODOs
- [x] Complete Clip.to_jams() function.
- [x] Add test_to_jams() function.
- [x] Complete documentation.

### CODE REVIEW

**Comment by magdalenafuentes** [comment]
Hey @pzinemanas I was going to start reviewing this but I noticed that the conflicts might affect my review, since I check the consistency and alphabetical order of loaders. Can you merge master and fix the conflicts and I'll review just after that?

Also sorry for taking so long!

**Comment by pzinemanas** [comment]
Hi @magdalenafuentes ! I've resolved the conflicts and merge the master into this branch. I think you can start reviewing now ;)

File: soundata/datasets/tau2020uas_mobile.py [hunk 1]
```diff
@@ -0,0 +1,961 @@
...
+"""
+
+import os
+from typing import BinaryIO, Optional, TextIO, Tuple
+
+import librosa
+import numpy as np
+import csv
+
+# import jams
```
Comment (magdalenafuentes) [inline soundata/datasets/tau2020uas_mobile.py]: remove

File: soundata/datasets/tau2020uas_mobile.py [hunk 2]
```diff
@@ -0,0 +1,961 @@
...
+                        metadata_index[clip_id] = {
+                            "scene_label": None,
+                            "city": None,
+                            "identifier": None,
+                            "source_label": None,
+                        }
+
+                    metadata_index[clip_id]["split"] = split
+
+        return metadata_index
```
Comment (magdalenafuentes) [inline soundata/datasets/tau2020uas_mobile.py:959]: Just to see if I follow: 1) you have a `metadata` file with all `clips` from the development subset, which you read first and fill each `field` of the metadata, but that file doesn't have the split information; 2) you get into the subsets of the `development` set and fill the split information for those, and add the `clip` ids of the evaluation?

**Review by magdalenafuentes — APPROVED** [review]
Great PR @pzinemanas! Just left a couple very minor comments, merge at convenience!

File: docs/source/soundata.rst [hunk 3]
```diff
@@ -27,6 +27,13 @@ TAU Urban Acoustic Scenes 2019
    :members:
    :inherited-members:
    
+TAU Urban Acoustic Scenes 2020 Mobile
```
Comment (justinsalamon) [inline docs/source/soundata.rst:30]: Can you add the dataset to the dataset list in the README?

File: tests/datasets/test_tau2020uas_mobile.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+
+def test_clip():
+    default_clipid = "airport-barcelona-0-0-a"
+    dataset = tau2020uas_mobile.Dataset(TEST_DATA_HOME)
+    clip = dataset.clip(default_clipid)
+
+    expected_attributes = {
+        "audio_path": (
+            "tests/resources/sound_datasets/tau2020uas_mobile/TAU-urban-acoustic-scenes-2020-mobile-development/audio/airport-barcelona-0-0-a.wav"
```
Comment (justinsalamon) [inline tests/datasets/test_tau2020uas_mobile.py:19]: Just to confirm, all test clips are <= 1s, yes?

**Review by justinsalamon — APPROVED** [review]
lgtm, see tiny comments below

File: tests/datasets/test_tau2020uas_mobile.py (hunk 4 above)
Comment (pzinemanas) [reply tests/datasets/test_tau2020uas_mobile.py:19]: Yes!

File: docs/source/soundata.rst (hunk 3 above)
Comment (pzinemanas) [reply docs/source/soundata.rst:30]: Yes!

File: soundata/datasets/tau2020uas_mobile.py (hunk 2 above)
Comment (pzinemanas) [reply soundata/datasets/tau2020uas_mobile.py:959]: Yes, but note that the step 2) includes splits from both development and evaluation datasets.

**Review by justinsalamon — APPROVED** [review]

**Comment by justinsalamon** [comment]
Awesome, unless @magdalenafuentes has any further comments I think this is ready! Once you merge this i'll open a version bump PR for `rc7` 🎉

**Comment by magdalenafuentes** [comment]
Go for it!

**Pull request merged**